systemtest/src/main/java/io/strimzi/systemtest/utils/StUtils.java

```diff
@@ -306,6 +306,13 @@ public static void waitForKafkaUserDeletion(String userName) {
         );
     }
 
+    public static void waitForKafkaClusterPodsDeletion(String clusterName) {
+        LOGGER.info("Waiting when all pods in Kafka cluster {} will be deleted", clusterName);
+        kubeClient().listPods().stream()
+                .filter(p -> p.getMetadata().getName().startsWith(clusterName))
+                .forEach(p -> StUtils.waitForPodDeletion(p.getMetadata().getName()));
```

Comment:
This seems like it could be flaky to me. You're relying on the pods being deleted in the same order that they're returned from the query. If they're deleted in a different order it's possible a pod will be deleted and recreated before you start waiting for it and so the wait will timeout.

Reply:
@tombentley I caught your thought. In this case, I do not expect the pod will be recreated. As for your case, I guess we need to use another method to check pod recreation. We have several methods for it: `waitForPodUpdate()`, `ssHasRolled()`, `waitTillSsHasRolled()` ... etc

Reply:
Thanks @serrss! It might be worth at least adding javadoc to make clear the intended use and point people to the alternatives for rolling.

Reply:
yeah, let me add javadoc for these methods